### PR TITLE
[BREAKING CHANGE] Enable various analyzer rules

### LINKED ISSUE #1411: Enable SA1616 rule
Change the action for SA1616 on the [ruleset file](https://github.com/kblok/puppeteer-sharp/blob/master/lib/PuppeteerSharp.ruleset) to Error, and make the solution build

### PR DESCRIPTION
**Pull request by tbaslyk**
Resolves #1411, #1409, #1405, #1403, #1402, #1401, #1400 and #1393. Everything is in one PR as it's mostly just formatting and style.

# Breaking change

* TargetType.iFrame is renamed to TargetType.IFrame

### CODE REVIEW

File: lib/PuppeteerSharp/BrowserFetcherOptions.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public class BrowserFetcherOptions
         /// </summary>
         /// <param name="address">address</param>
         /// <param name="fileName">fileName</param>
-        /// <returns></returns>
+        /// <returns>Task</returns>
```
Comment (kblok) [inline lib/PuppeteerSharp/BrowserFetcherOptions.cs:16]:
```suggestion
        /// <returns>A Task that resolves when the download finishes.</returns>
```

File: lib/PuppeteerSharp/LauncherBase.cs [hunk 2]
```diff
@@ -108,23 +103,23 @@ public void Dispose()
         /// <summary>
         /// Asynchronously starts Base process.
         /// </summary>
-        /// <returns></returns>
+        /// <returns>Task</returns>
```
Comment (kblok) [inline lib/PuppeteerSharp/LauncherBase.cs:106]:
```suggestion
        /// <returns>Task which resolves when after start process begins.</returns>
```

File: lib/PuppeteerSharp/Page.cs [hunk 3]
```diff
@@ -929,7 +929,7 @@ public Task<Response> GoToAsync(string url, WaitUntilNavigation waitUntil)
         /// generates a pdf of the page with <see cref="MediaType.Print"/> css media. To generate a pdf with <see cref="MediaType.Screen"/> media call <see cref="EmulateMediaAsync(MediaType)"/> with <see cref="MediaType.Screen"/>
         /// </summary>
         /// <param name="file">The file path to save the PDF to. paths are resolved using <see cref="Path.GetFullPath(string)"/></param>
-        /// <returns></returns>
+        /// <returns>Task</returns>
```
Comment (kblok) [inline lib/PuppeteerSharp/Page.cs:932]:
```suggestion
        /// <returns>A Task which resolves after the PDF is generated</returns>
```

File: lib/PuppeteerSharp/Page.cs [hunk 4]
```diff
@@ -940,7 +940,7 @@ public Task<Response> GoToAsync(string url, WaitUntilNavigation waitUntil)
         /// </summary>
         /// <param name="file">The file path to save the PDF to. paths are resolved using <see cref="Path.GetFullPath(string)"/></param>
         /// <param name="options">pdf options</param>
-        /// <returns></returns>
+        /// <returns>Task</returns>
```
Comment (kblok) [inline lib/PuppeteerSharp/Page.cs:943]:
```suggestion
        /// <returns>A Task which resolves after the PDF is generated</returns>
```

File: lib/PuppeteerSharp/Page.cs [hunk 5]
```diff
@@ -1082,7 +1082,7 @@ public Task SetJavaScriptEnabledAsync(bool enabled)
         /// Toggles bypassing page's Content-Security-Policy.
         /// </summary>
         /// <param name="enabled">sets bypassing of page's Content-Security-Policy.</param>
-        /// <returns></returns>
+        /// <returns>Task</returns>
```
Comment (kblok) [inline lib/PuppeteerSharp/Page.cs:1085]:
```suggestion
        /// <returns>Task which resolves after the message is sent to the browser</returns>
```

File: lib/PuppeteerSharp/Page.cs [hunk 6]
```diff
@@ -1547,7 +1547,7 @@ public Task SetExtraHttpHeadersAsync(Dictionary<string, string> headers)
         /// Provide credentials for http authentication <see href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication"/>
         /// </summary>
         /// <param name="credentials">The credentials</param>
-        /// <returns></returns>
+        /// <returns>Task</returns>
```
Comment (kblok) [inline lib/PuppeteerSharp/Page.cs:1550]:
```suggestion
        /// <returns>Task which resolves after the message is sent to the browser</returns>
```

**Review by kblok — COMMENTED** [review]
Thank you!
Just a small request :)

File: lib/PuppeteerSharp/Puppeteer.cs [hunk 7]
```diff
@@ -26,14 +26,14 @@ public static class Puppeteer
         /// <summary>
         /// The default flags that Chromium will be launched with.
         /// </summary>
-        internal static string[] DefaultArgs => ChromiumLauncher.DefaultArgs;
+        internal static string[] DefaultArgs => ChromiumLauncher._defaultArgs;
```
[the remark this answers is not in the document]
Comment (tbaslyk) [reply lib/PuppeteerSharp/Puppeteer.cs:29]: Wait I'm a bit confused, isn't this already an internal prop that encapsulates field?

[the remark this answers is not in the document]
Comment (kblok) [reply lib/PuppeteerSharp/Puppeteer.cs:29]: I'm talking about `ChromiumLauncher._defaultArgs`. We could turn it into a read-only property called `ChromiumLauncher.DefaultArgs`.

[the remark this answers is not in the document]
Comment (tbaslyk) [reply lib/PuppeteerSharp/Puppeteer.cs:29]: Ah, sure, that makes more sense!

**Review by kblok — APPROVED** [review]
Thank you!

**Pull request merged**